src/Bicep.Core/CodeAction/CodeReplacement.cs

```diff
@@ -15,5 +18,14 @@ public CodeReplacement(TextSpan span, string text)
         public TextSpan Span { get; }
 
         public string Text { get; }
+
+        public static CodeReplacement FromSyntax(TextSpan span, SyntaxBase syntax)
+        {
+            var sb = new StringBuilder();
+            var documentBuildVisitor = new DocumentBuildVisitor();
+            var document = documentBuildVisitor.BuildDocument(syntax);
+            document.Layout(sb, "", System.Environment.NewLine);
```

Comment:
>Layout [](http://example.com/codeflow?start=21&length=6)

The functionality to format the current node as a string could be useful in other places. Is it worth moving that part to `SyntaxBase` or an extension method?

(One example would be including a full expression string in an error message.)